### DISCUSSION TITLE
Moduliths to become a Spring experimental project

### DISCUSSION THREAD

**Discussion by odrotbohm** [odrotbohm/moduliths#246] [Announcements]
I'm delighted to announce that _Moduliths_ is going to become an official, Spring experimental project. [_Spring Modulith_](https://github.com/spring-projects-experimental/spring-modulith) will be based on Spring Boot 3.0 and Java 17, and we will use the move to reset the version number to 0.1, both to be able to revisit some default settings and also express the incubating status of the project. The current plan is to release 0.1 GA right after Spring Boot 3.0 GA, currently scheduled for the end of November, with a couple of milestones and release candidates in between.

Users of _Moduliths_ can give the `0.1.0-SNAPSHOT` of Spring Modulith (available via the [Spring snapshot repository](https://repo.spring.io/ui/native/snapshot/org/springframework/experimental/)) a try by changing the artifact coordinates to `org.springframework.experimental:spring-modulith-…` (note the lack of the trailing "s"!). We have already changed [a few of the core abstractions](https://github.com/spring-projects-experimental/spring-modulith/issues/9), primarily renaming everything that was `Module…` before to `ApplicationModule`, both for the optional annotations and the core model abstractions to be used in test code. We're going to ship a detailed upgrade guide with the first milestone of _Spring Modulith_ 0.1.

_Moduliths_ will be maintained on its current 1.3 generation until Spring Boot 2.7 goes out of open-source maintenance (currently scheduled for November 2023). We don't plan to add any significant new features, but would be willing to help the community port ones that we add on the _Spring Modulith_ side if there is significant interest.

**Comment by vpavic** [comment]
Congrats! :slightly_smiling_face:

To make this news a bit more visible to people interacting with this repo, it might be a good idea to update the readme on `main` branch with a link to new project home and some additional info (potentially link to this announcement).

**Reply by odrotbohm** [reply]
Sure thing! We will have more information updated once we announce the initial release of the project on the Spring side of thing.

**Reply by odrotbohm** [reply]
The project migration note has been added to [the readme](https://github.com/moduliths/moduliths#moduliths).
